### PR TITLE
WIP in progress deployment

### PR DESCRIPTION
**Pull request by fabianofranz**

### CODE REVIEW

**Comment by fabianofranz** [comment]
@spadgett @jwforres

File: app/scripts/directives/overviewService.js [hunk 1]
```diff
@@ -48,14 +49,39 @@ angular.module('openshiftConsole')
           return isRecentDeployment(deployment, dc);
         };
 
+        $scope.visibleDeployments = function(deployments) {
+          return _.filter(deployments, $scope.isDeploymentVisible);
+        };
+
+        $scope.deploymentInProgress = function(deployments) {
+          return anyDeploymentIsInProgress(deployments);
```
Comment (spadgett) [inline app/scripts/directives/overviewService.js]: I'd just use the filter directly from the view.

File: app/scripts/directives/overviewService.js [hunk 2]
```diff
@@ -48,14 +49,39 @@ angular.module('openshiftConsole')
...
+        };
+
+        $scope.isDeploymentLatest = function(deployment) {
```
Comment (spadgett) [inline app/scripts/directives/overviewService.js:55]: We have a `deploymentIsLatest` filter you should be able to use.

https://github.com/spadgett/origin-web-console/blob/e32941237fb8e9fbc76a932491f4d56d2455b31e/app/scripts/filters/resources.js#L778

**Comment by fabianofranz** [comment]
@spadgett conflicts fixed and addressed comments except for `isDeploymentLatest`: the difference in comparison to the filter is that I don't have the `deploymentConfig` so I traverse the list of dcs to find it.

I'll submit a separate PR to what's missing in the task (mostly handling failure scenarios) and a few improvements I want to do to the code.

**Pull request merged**